src/app/modules/did-book/components/select-did/select-did.component.ts

```diff
@@ -1,54 +1,104 @@
-import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 import { FormControl, Validators } from '@angular/forms';
-import { HexValidators } from '../../../../utils/validators/is-hex/is-hex.validator';
+import { HexValidators } from '@utils';
 import { DidBookService } from '../../services/did-book.service';
-import { combineLatest } from 'rxjs';
+import { combineLatest, Observable } from 'rxjs';
 import { map, startWith } from 'rxjs/operators';
 import { MatDialog } from '@angular/material/dialog';
 import { AddSingleRecordComponent } from '../add-single-record/add-single-record.component';
+import { ListValidator } from '@utils';
+import { DidBookRecord } from '../models/did-book-record';
 
 @Component({
   selector: 'app-select-did',
   templateUrl: './select-did.component.html',
   styleUrls: ['./select-did.component.scss'],
 })
 export class SelectDidComponent implements OnInit {
+  @Input() label: string;
+  @Input() placeholder: string;
+  @Input() isRequired: boolean = false;
+  @Input() showAddButton: boolean;
+  @Input() set didToRemove(list: string[]) {
+    this._didToRemove = list;
+    this.setValidators();
+    this.setDIDBookList();
+  }
+  get didToRemove() {
+    return this._didToRemove;
+  }
   @Output() didChange = new EventEmitter<{ did: string; valid: boolean }>();
+  @Output() add = new EventEmitter<string>();
 
-  didBook$;
-  newOwnerDID = new FormControl('', [
-    Validators.required,
-    HexValidators.isDidValid(),
-  ]);
-  isNotKnownDid: boolean;
+  didBook$: Observable<DidBookRecord[]>;
+  newDID = new FormControl('', [HexValidators.isDidValid()]);
+  isNotKnownDid: boolean = false;
+  private _didToRemove: string[] = [];
```

Comment:
Sometimes I've seen that private class fields should come before public ones when being declared. But I think this is a hold over from my C# days and not applicable to Angular 😄 https://stackoverflow.com/a/310967

Reply:
I do not know if are there any good practices for JS/TS/angular about private/public fields. most of the times I'm writing first public second private. Anyway in the end of the day it doesn't matter in JS world, because all properties are public.